test_runner/regress/test_branching.py

```diff
@@ -347,6 +347,70 @@ def start_creating_timeline():
         ps_http.timeline_detail(env.initial_tenant, branch_id)
 
 
+def test_duplicate_creation(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_configs()
+    env.start()
+    env.pageserver.tenant_create(env.initial_tenant)
+
+    success_timeline = TimelineId.generate()
+    log.info(f"Creating timeline {success_timeline}")
+    ps_http = env.pageserver.http_client()
+    success_result = ps_http.timeline_create(
+        env.pg_version, env.initial_tenant, success_timeline, timeout=60
+    )
+
+    ps_http.configure_failpoints(("timeline-creation-after-uninit", "pause"))
+    ps_http.configure_failpoints(("before-upload-index-pausable", "pause"))
+    ps_http.configure_failpoints(("before-upload-layer-pausable", "pause"))
+
+    def start_creating_timeline():
+        log.info(f"Creating (expect failure) timeline {env.initial_timeline}")
+        with pytest.raises(RequestException):
+            ps_http.timeline_create(
+                env.pg_version, env.initial_tenant, env.initial_timeline, timeout=60
+            )
+
+    t = threading.Thread(target=start_creating_timeline)
+    try:
+        t.start()
+
+        wait_until_paused(env, "timeline-creation-after-uninit")
+
+        # While in this "creation hung" state we will validate behavior of concurrent requests
+        # env.pageserver.allowed_errors.append(
+        #    ".*request{method=POST.*.*"
+        # )
+        # While timeline creation is in progress, trying to create a timeline
+        # again with the same ID should return 409
```

Comment:
Feels like either one of these comments and commented out code is extra.

Reply:
Yep, cleaned up in 0e03700a928cf55eca8b851c9148fb1ee68e7f19